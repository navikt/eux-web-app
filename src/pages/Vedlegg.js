import React, { Component } from 'react';
import PT from 'prop-types';
import { change, formValueSelector, reduxForm } from 'redux-form';
import { connect } from 'react-redux';

import * as Skjema from 'felles-komponenter/skjema';
import { StatusLinje } from '../felles-komponenter/statuslinje';
import DokumentSok from './dokumentsok';
import Ui from 'eessi-pensjon-ui'
import * as MPT from '../proptypes';

import { KodeverkSelectors } from '../ducks/kodeverk';
import { RinavedleggOperations, RinavedleggSelectors } from '../ducks/rinavedlegg';
import { getParam } from '../utils/queryString';
import './Vedlegg.css';

class Vedlegg extends Component {
  componentDidMount() {
    const { location, oppdaterRinaSaksnummer } = this.props;
    const rinasaksnummer = getParam(location, 'rinasaksnummer');
    oppdaterRinaSaksnummer(rinasaksnummer);
  }

  overrideDefaultSubmit = event => {
    event.preventDefault();
  };
  render() {
    const {
      handleSubmit, sendSkjema, vedleggStatus, vedlegg, inntastetRinasaksnummer, rinadokumentID, settRinaGyldighet, settRinaSjekket, rinaNrErGyldig, rinaNrErSjekket,
    } = this.props;
    const responsLenke = vedlegg && vedlegg.url;
    const visVenteSpinner = ['PENDING'].includes(vedleggStatus);
    const disableSendKnapp = !(rinaNrErGyldig && rinaNrErSjekket && rinadokumentID);

    return (
      <div className="vedlegg">
        <Ui.Nav.Container fluid>
          <form onSubmit={this.overrideDefaultSubmit}>
            <Ui.Nav.Row>
              <Ui.Nav.Column xs="6">
                <Ui.Nav.Panel className="vedlegg__skjema">
                  <Ui.Nav.Fieldset legend="Vedleggs informasjon">
                    <Ui.Nav.HjelpetekstBase id="journalPostID" type="hoyre">Journalpost ID finner du i Gosys</Ui.Nav.HjelpetekstBase>
                    <Skjema.Input feltNavn="journalpostID" label="JournalpostID" />
                    <Ui.Nav.HjelpetekstBase id="dokumentID" type="under">Dokument ID finner du i Gosys</Ui.Nav.HjelpetekstBase>
                    <Skjema.Input feltNavn="dokumentID" label="DokumentID" />
                    <DokumentSok
                      inntastetRinasaksnummer={inntastetRinasaksnummer}
                      settRinaGyldighet={settRinaGyldighet}
                      settRinaSjekket={settRinaSjekket}
                    />
                  </Ui.Nav.Fieldset>
                  <div className="vedlegg__submmit">
                    <Ui.Nav.Hovedknapp
                      onClick={handleSubmit(sendSkjema)}
                      disabled={disableSendKnapp || visVenteSpinner}
                      spinner={visVenteSpinner}>Send vedlegg
                    </Ui.Nav.Hovedknapp>
                  </div>
                  <StatusLinje status={vedleggStatus} url={responsLenke} tittel="Vedlegget" />
                </Ui.Nav.Panel>
              </Ui.Nav.Column>
            </Ui.Nav.Row>
          </form>
        </Ui.Nav.Container>
      </div>
    );
  }
}

Vedlegg.propTypes = {
  location: PT.object.isRequired,
  sedtyper: PT.arrayOf(MPT.Kodeverk),
  rinaNrErGyldig: PT.bool,
  rinaNrErSjekket: PT.bool,
  vedlegg: PT.object,
  rinasaksnummer: PT.string,
  inntastetRinasaksnummer: PT.string,
  rinadokumentID: PT.string,
  // Funcs
  settRinaGyldighet: PT.func.isRequired,
  settRinaSjekket: PT.func.isRequired,
  vedleggStatus: PT.string.isRequired,
  handleSubmit: PT.func.isRequired,
  sendSkjema: PT.func.isRequired,
  oppdaterRinaSaksnummer: PT.func.isRequired,
};

Vedlegg.defaultProps = {
  sedtyper: undefined,
  vedlegg: {},
  rinasaksnummer: '',
  inntastetRinasaksnummer: '',
  rinadokumentID: '',
  rinaNrErGyldig: false,
  rinaNrErSjekket: false,
};

const skjemaSelector = formValueSelector('vedlegg');

const mapStateToProps = state => ({
  vedleggStatus: RinavedleggSelectors.vedleggStatusSelector(state),
  vedlegg: RinavedleggSelectors.vedleggSelector(state),
  sedtyper: KodeverkSelectors.alleSEDtyperSelector(state),
  inntastetRinasaksnummer: skjemaSelector(state, 'rinasaksnummer'),
  rinadokumentID: skjemaSelector(state, 'rinadokumentID'),
  rinaNrErGyldig: skjemaSelector(state, 'rinaNrErGyldig'),
  rinaNrErSjekket: skjemaSelector(state, 'rinaNrErSjekket'),
});

const mapDispatchToProps = dispatch => ({
  sendSkjema: data => dispatch(RinavedleggOperations.sendVedlegg(data)),
  oppdaterRinaSaksnummer: rinasaksnummer => dispatch(change('vedlegg', 'rinasaksnummer', rinasaksnummer)),
  settRinaGyldighet: erGyldig => dispatch(change('vedlegg', 'rinaNrErGyldig', erGyldig)),
  settRinaSjekket: erSjekket => dispatch(change('vedlegg', 'rinaNrErSjekket', erSjekket)),
});

const journalpostValidation = journalpostID => {
  if (!journalpostID) { return 'Du m책 taste inn en journalpostID'; }
  return null;
};

const form = {
  form: 'vedlegg',
  enableReinitialize: true,
  destroyOnUnmount: true,
  onSubmit: () => {},
  validate: values => {
    const journalpostID = journalpostValidation(values.journalpostID);
    const dokumentID = !values.dokumentID ? 'Du m책 taste inn en dokumentID' : null;
    const saksnummer = !values.saksnummer ? 'Du m책 taste inn et RINA saksnummer' : null;
    const rinadokumentID = !values.rinadokumentID ? 'Du m책 velge en SED' : null;
    return {
      journalpostID,
      dokumentID,
      saksnummer,
      rinadokumentID,
    };
  },
};
export default connect(mapStateToProps, mapDispatchToProps)(reduxForm(form)(Vedlegg));
