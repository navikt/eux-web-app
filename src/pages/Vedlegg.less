.vedlegg {
  width: 100%;

  .hjelpetekst {
    float: right;
  }

  .vedlegg__lenke {
    margin-top: 2em;
  }

  .vedlegg_hjelpeikon {

  }
}
