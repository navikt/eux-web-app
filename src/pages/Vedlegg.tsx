import * as vedleggActions from 'actions/vedlegg'
import DocumentSearch from 'components/DocumentSearch/DocumentSearch'
import { Container, Content, Margin, VerticalSeparatorDiv } from 'components/StyledComponents'
import TopContainer from 'components/TopContainer/TopContainer'
import { State } from 'declarations/reducers'
import { Validation } from 'declarations/types'
import _ from 'lodash'
import AlertStripe from 'nav-frontend-alertstriper'
import Hjelpetekst from 'nav-frontend-hjelpetekst'
import { Hovedknapp } from 'nav-frontend-knapper'
import Lenke from 'nav-frontend-lenker'
import { FeiloppsummeringFeil, Input } from 'nav-frontend-skjema'
import { Systemtittel } from 'nav-frontend-typografi'
import PT from 'prop-types'
import React, { useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { useDispatch, useSelector } from 'react-redux'
import styled from 'styled-components'

export interface VedleggSelector {
  vedlegg: any;
  rinasaksnummer: any;
  journalpostID: any;
  rinadokumentID: any;
  dokumentID: any;
  sendingVedlegg: boolean;
}

export interface VedleggProps {
  location: any;
}

const mapState = (state: State): VedleggSelector => ({
  vedlegg: state.vedlegg.vedlegg,
  rinasaksnummer: state.vedlegg.rinasaksnummer,
  journalpostID: state.vedlegg.journalpostID,
  rinadokumentID: state.vedlegg.rinadokumentID,
  dokumentID: state.vedlegg.dokumentID,
  sendingVedlegg: state.loading.sendingVedlegg
})

const Link = styled(Lenke)`
  margin-top: 2em;
`

const Vedlegg: React.FC<VedleggProps> = ({ location }: VedleggProps): JSX.Element => {
  const [mounted, setMounted] = useState(false)
  const dispatch = useDispatch()
  const { t } = useTranslation()
  const { journalpostID, dokumentID, rinasaksnummer, rinadokumentID, sendingVedlegg, vedlegg }: VedleggSelector = useSelector<State, VedleggSelector>(mapState)
  const [validation, setValidation] = useState<Validation>({})
  const [isRinaNumberValid, setIsRinaNumberValid] = useState<boolean>(false)

  useEffect(() => {
    if (!mounted) {
      const params: URLSearchParams = new URLSearchParams(location.search)
      const rinasaksnummer = params.get('rinasaksnummer')
      if (rinasaksnummer) {
        dispatch(vedleggActions.propertySet('rinasaksnummer', rinasaksnummer))
      }
      setMounted(true)
    }
  }, [mounted, dispatch, location])

  const isValid = (_validation: Validation): boolean => {
    return _.find(_.values(_validation), e => e !== undefined) === undefined
  }

  const validate = (): Validation => {
    const validation = {
      journalpostID: !journalpostID ? {
        feilmelding: t('ui:validation-noJournalpostID'),
        skjemaelementId: 'vedlegg-journalpostID-id'
      } as FeiloppsummeringFeil : undefined,
      dokumentID: !dokumentID ? {
        feilmelding: t('ui:validation-noDokumentID'),
        skjemaelementId: 'vedlegg-dokumentID-id'
      } : undefined,
      rinasaksnummer: !rinasaksnummer ? {
        feilmelding: t('ui:validation-noSaksnummer'),
        skjemaelementId: ''
      } : (!isRinaNumberValid ? {
        feilmelding: t('ui:validation-unverifiedSaksnummer'),
        skjemaelementId: ''
      } : undefined),
      rinadokumentID: !rinadokumentID ? {
        feilmelding: t('ui:validation-noRinadokumentID'),
        skjemaelementId: ''
      } : undefined
    }
    setValidation(validation)
    return validation
  }

  const resetValidation = (key: string): void => {
    setValidation({
      ...validation,
      [key]: undefined
    })
  }

  const sendSkjema = (): void => {
    if (isValid(validate()) && isRinaNumberValid) {
      dispatch(vedleggActions.sendVedlegg({
        journalpostID: journalpostID,
        dokumentID: dokumentID,
        rinasaksnummer: rinasaksnummer,
        rinadokumentID: rinadokumentID,
        rinaNrErSjekket: isRinaNumberValid,
        rinaNrErGyldig: isRinaNumberValid
      }))
    }
  }

  const onjournalpostIDChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
    resetValidation('journalpostID')
    dispatch(vedleggActions.propertySet('journalpostID', e.target.value))
  }

  const onDokumentIDChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
    resetValidation('dokumentID')
    dispatch(vedleggActions.propertySet('dokumentID', e.target.value))
  }

  return (
    <TopContainer>
      <Container>
        <Margin />
        <Content>
          <VerticalSeparatorDiv />
          <Systemtittel>
            {t('ui:title-vedlegg')}
          </Systemtittel>
          <VerticalSeparatorDiv />
          <div className='noSlideAnimate'>
            <Input
              id='vedlegg-journalpostID-id'
              data-test-id='vedlegg-journalpostID-id'
              label={(
                <div>
                  {t('ui:label-journalpostID')}
                  <Hjelpetekst id='journalPostID'>
                    {t('ui:help-journalpostID')}
                  </Hjelpetekst>
                </div>
              )}
              onChange={onjournalpostIDChange}
              feil={validation.journalpostID}
            />
          </div>
          <VerticalSeparatorDiv />
          <div className='noSlideAnimate' style={{ animationDelay: '0.15s' }}>
            <Input
              id='vedlegg-dokumentID-id'
              data-test-id='vedlegg-dokumentID-id'
              label={(
                <div>
                  {t('ui:label-dokumentID')}
                  <Hjelpetekst id='dokumentID'>
                    {t('ui:help-dokumentID')}
                  </Hjelpetekst>
                </div>
              )}
              onChange={onDokumentIDChange}
              feil={validation.dokumentID}
            />
            <VerticalSeparatorDiv />
          </div>
          <div className='noSlideAnimate' style={{ animationDelay: '0.0s' }}>
            <DocumentSearch
              validation={validation}
              resetValidation={resetValidation}
              onRinasaksnummerChanged={() => setIsRinaNumberValid(false)}
              onDocumentFound={() => setIsRinaNumberValid(true)}
            />
            <VerticalSeparatorDiv />
          </div>
          <div className='slideAnimate' style={{ animationDelay: '0.45s' }}>
            <Hovedknapp
              onClick={sendSkjema}
              disabled={sendingVedlegg}
              spinner={sendingVedlegg}
            >
              {sendingVedlegg ? t('ui:label-sendingVedlegg') : t('ui:label-sendVedlegg')}
            </Hovedknapp>
            {vedlegg && (
              <>
                <VerticalSeparatorDiv />
                <AlertStripe type='suksess'>
                  <div>
                    <div>Vedlegget: {vedlegg.filnavn || vedlegg.vedleggID}</div>
                    {vedlegg.url && (
                      <Link href={vedlegg.url} target='_blank'>
                        GÃ¥ direkte til Rina.
                      </Link>
                    )}
                  </div>
                </AlertStripe>
              </>
            )}
          </div>
        </Content>
        <Margin />
      </Container>
    </TopContainer>
  )
}

Vedlegg.propTypes = {
  location: PT.object.isRequired
}

export default Vedlegg
