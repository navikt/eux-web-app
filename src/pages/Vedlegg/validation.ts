import { Validation } from 'declarations/types'
import { TFunction } from 'react-i18next'
import { checkIfNotEmpty } from 'utils/validation'

export interface ValidationVedleggProps {
  journalpostID: string | undefined
  dokumentID: string | undefined
  rinasaksnummer: string | undefined
  rinadokumentID: string | undefined
  namespace: string
}

export const validateVedlegg = (
  v: Validation,
  t: TFunction,
  {
    journalpostID,
    dokumentID,
    rinasaksnummer,
    rinadokumentID,
    namespace
  }: ValidationVedleggProps
): boolean => {
  const hasErrors: Array<boolean> = []

  hasErrors.push(checkIfNotEmpty(v, {
    needle: journalpostID,
    id: namespace + '-journalpostID',
    message: 'validation:noJournalpostID'
  }))

  hasErrors.push(checkIfNotEmpty(v, {
    needle: dokumentID,
    id: namespace + '-dokumentID',
    message: 'validation:noDokumentID'
  }))

  hasErrors.push(checkIfNotEmpty(v, {
    needle: rinasaksnummer,
    id: namespace + '-documentSearch-rinasaksnummer',
    message: 'validation:noSaksnummer'
  }))

  hasErrors.push(checkIfNotEmpty(v, {
    needle: rinadokumentID,
    id: namespace + '-documentSearch-rinadokumentID',
    message: 'validation:noRinadokumentID'
  }))

  return hasErrors.find(value => value) !== undefined
}
