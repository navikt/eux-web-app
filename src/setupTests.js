/* eslint-disable */

import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import each from 'jest-each';

global.fetch = require('jest-fetch-mock');

Enzyme.configure({ adapter: new Adapter() });

global.shallow = shallow;
global.mount = mount;

global.each = each;
