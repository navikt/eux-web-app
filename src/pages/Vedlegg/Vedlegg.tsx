import { Alert, Button, Checkbox, HelpText, Link, Loader } from '@navikt/ds-react'
import {
  AlignStartRow,
  Column,
  Container,
  Content,
  FlexDiv,
  HorizontalSeparatorDiv,
  Margin,
  VerticalSeparatorDiv
} from '@navikt/hoykontrast'
import { resetValidation, setValidation } from 'actions/validation'
import * as vedleggActions from 'actions/vedlegg'
import DocumentSearch from 'applications/Vedlegg/DocumentSearch/DocumentSearch'
import Input from 'components/Forms/Input'
import TopContainer from 'components/TopContainer/TopContainer'
import ValidationBox from 'components/ValidationBox/ValidationBox'
import * as types from 'constants/actionTypes'
import { State } from 'declarations/reducers'
import { Validation, VedleggPayload, VedleggSendResponse } from 'declarations/types'
import _ from 'lodash'
import performValidation from 'utils/performValidation'
import React, { useEffect } from 'react'
import { useTranslation } from 'react-i18next'
import { useAppDispatch, useAppSelector } from 'store'
import styled from 'styled-components'
import { validateVedlegg, ValidationVedleggProps } from './validation'

export interface VedleggSelector {
  alertMessage: JSX.Element | string | undefined
  alertType: string | undefined
  dokumentID: string | undefined
  journalpostID: string | undefined
  rinadokumentID: string | undefined
  rinasaksnummer: string | undefined
  sensitivt: boolean
  sendingVedlegg: boolean
  vedleggResponse: VedleggSendResponse | undefined
  validation: Validation
}

const mapState = (state: State): VedleggSelector => ({
  alertMessage: state.alert.stripeMessage,
  alertType: state.alert.type,
  sendingVedlegg: state.loading.sendingVedlegg,
  dokumentID: state.vedlegg.dokumentID,
  journalpostID: state.vedlegg.journalpostID,
  rinadokumentID: state.vedlegg.rinadokumentID,
  rinasaksnummer: state.vedlegg.rinasaksnummer,
  sensitivt: state.vedlegg.sensitivt,
  vedleggResponse: state.vedlegg.vedleggResponse,
  validation: state.validation.status
})

export const MyContent = styled(Content)`
  @media (min-width: 768px) {
   min-width: 600px;
   maxWidth: 800px;
 }
  align-items: center;
`

const Vedlegg: React.FC = (): JSX.Element => {
  const dispatch = useAppDispatch()
  const namespace = 'vedlegg'
  const { t } = useTranslation()
  const { alertMessage, alertType, journalpostID, dokumentID, rinasaksnummer, rinadokumentID, sendingVedlegg, sensitivt, vedleggResponse, validation }: VedleggSelector = useAppSelector(mapState)

  useEffect(() => {
    const params: URLSearchParams = new URLSearchParams(window.location.search)
    const rinasaksnummer = params.get('rinasaksnummer')
    if (rinasaksnummer) {
      dispatch(vedleggActions.propertySet('rinasaksnummer', rinasaksnummer))
    }
  }, [])

  const sendSkjema = () => {
    const clonedValidation = _.cloneDeep(validation)
    const hasErrors = performValidation<ValidationVedleggProps>(clonedValidation, namespace, validateVedlegg, {
      journalpostID,
      dokumentID,
      rinasaksnummer,
      rinadokumentID
    })
    dispatch(setValidation(clonedValidation))
    if (!hasErrors) {
      dispatch(vedleggActions.sendVedlegg({
        journalpostID,
        dokumentID,
        rinasaksnummer,
        rinadokumentID,
        sensitivt
      } as VedleggPayload))
    }
  }

  const setJournalpostID = (newJournalpostID: string): void => {
    dispatch(vedleggActions.propertySet('journalpostID', newJournalpostID))
    if (validation[namespace + '-journalpostID']) {
      dispatch(resetValidation(namespace + '-journalpostID'))
    }
  }

  const setDokumentID = (newDokumentID: string): void => {
    dispatch(vedleggActions.propertySet('dokumentID', newDokumentID))
    if (validation[namespace + '-dokumentID']) {
      dispatch(resetValidation(namespace + '-dokumentID'))
    }
  }

  const setSensitivt = (newSensitivt: boolean): void => {
    dispatch(vedleggActions.propertySet('sensitivt', newSensitivt))
    if (validation[namespace + '-sensitivt']) {
      dispatch(resetValidation(namespace + '-sensitivt'))
    }
  }

  return (
    <TopContainer title={t('app:page-title-vedlegg')}>
      <Container>
        <Margin />
        <MyContent>
          <AlignStartRow>
            <Column>
              <Input
                id='journalpostID'
                data-testid={namespace + '-journalpostID'}
                namespace={namespace}
                value={journalpostID}
                label={(
                  <FlexDiv>
                    {t('label:journalpost-id') + ' *'}
                    <HorizontalSeparatorDiv size='0.35' />
                    <HelpText id='journalPostID'>
                      {t('message:help-journalpostID')}
                    </HelpText>
                  </FlexDiv>
                )}
                onChanged={setJournalpostID}
                error={validation[namespace + '-journalpostID']?.feilmelding}
              />
              <VerticalSeparatorDiv />
            </Column>
            <Column>
              <Input
                id='dokumentID'
                namespace={namespace}
                data-testid={namespace + '-dokumentID'}
                value={dokumentID}
                label={(
                  <FlexDiv>
                    {t('label:dokument-id') + ' *'}
                    <HorizontalSeparatorDiv size='0.35' />
                    <HelpText id={namespace + '-dokumentID-help'}>
                      {t('message:help-dokumentID')}
                    </HelpText>
                  </FlexDiv>
                )}
                onChanged={setDokumentID}
                error={validation[namespace + '-dokumentID']?.feilmelding}
              />
              <VerticalSeparatorDiv />
            </Column>
          </AlignStartRow>
          <AlignStartRow>
            <Column>
              <Checkbox
                checked={sensitivt}
                data-testid={namespace + '-sensitivt'}
                error={!!validation[namespace + '-sensitivt']?.feilmelding}
                onChange={(e: React.ChangeEvent<HTMLInputElement>) => setSensitivt(e.target.checked)}
              >
                {t('label:sensitivt')}
              </Checkbox>
              <VerticalSeparatorDiv />
            </Column>
          </AlignStartRow>
          <AlignStartRow>
            <DocumentSearch
              parentNamespace={namespace}
              validation={validation}
              resetValidation={resetValidation}
            />
          </AlignStartRow>
          <VerticalSeparatorDiv size='2' />
          <AlignStartRow>
            <Column>
              <Button
                variant='primary'
                onClick={sendSkjema}
                disabled={sendingVedlegg}
              >
                {sendingVedlegg ? t('message:loading-sending-vedlegg') : t('label:send-vedlegg')}
                {sendingVedlegg && <Loader />}
              </Button>
            </Column>
          </AlignStartRow>
          <VerticalSeparatorDiv size='2' />
          {alertMessage && alertType && [types.VEDLEGG_POST_FAILURE].indexOf(alertType) >= 0 && (
            <>
              <Alert variant='warning'>
                {alertMessage}
              </Alert>
              <VerticalSeparatorDiv size='2' />
            </>
          )}
          <ValidationBox
            heading={t('validation:feiloppsummering')}
            validation={validation}
          />
          <VerticalSeparatorDiv size='2' />
          {vedleggResponse && (
            <Alert variant='success'>
              <div>
                <div>{t('label:vedlagte')}: {vedleggResponse.filnavn || vedleggResponse.vedleggID}</div>
                {vedleggResponse.url && (
                  <Link href={vedleggResponse.url} rel='noreferrer' target='_blank'>
                    {t('label:g√•-til-rina')}
                  </Link>
                )}
              </div>
            </Alert>
          )}
        </MyContent>
        <Margin />
      </Container>
    </TopContainer>
  )
}

export default Vedlegg
