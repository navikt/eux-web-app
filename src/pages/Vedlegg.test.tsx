import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import { stageSelector } from 'setupTests'
import Vedlegg, { VedleggProps, VedleggSelector } from './Vedlegg'

jest.mock('react-router-dom', () => ({
  Link: () => (<div className='mock-link' />)
}))
jest.mock('eessi-pensjon-ui', () => {
  const Ui = jest.requireActual('eessi-pensjon-ui').default
  return {
    ...Ui,
    Nav: {
      ...Ui.Nav,
      Hjelpetekst: ({ children }: {children: any}) => <div className='mock-hjelpetekst'>{children}</div>
    }
  }
})

describe('pages/Vedlegg', () => {
  let wrapper: ReactWrapper
  const initialMockProps: VedleggProps = {
    location: {
      host: '',
      hash: '',
      hostname: '',
      href: '',
      origin: '',
      pathname: '',
      port: '',
      protocol: '',
      search: ''
    } as Location
  }

  const defaultSelector: VedleggSelector = {
    vedlegg: undefined,
    rinasaksnummer: undefined,
    journalpostID: undefined,
    rinadokumentID: undefined,
    dokumentID: undefined,
    sendingVedlegg: false
  }

  beforeAll(() => {
    stageSelector(defaultSelector, {})
  })

  beforeEach(() => {
    wrapper = mount(<Vedlegg {...initialMockProps} />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Renders', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('HTML structure', () => {
    expect(wrapper.exists('.vedlegg')).toBeTruthy()
  })
})
