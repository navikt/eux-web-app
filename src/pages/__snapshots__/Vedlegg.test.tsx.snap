// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['pages/Vedlegg Renders 1'] = `
<Vedlegg
  location={
    Object {
      "hash": "",
      "host": "",
      "hostname": "",
      "href": "",
      "origin": "",
      "pathname": "",
      "port": "",
      "protocol": "",
      "search": "",
    }
  }
>
  <TopContainer
    className="vedlegg"
  >
    <UseErrorBoundaryWrapper
      renderError={[Function]}
    >
      <ErrorBoundary
        onDidCatch={[Function]}
        renderError={[Function]}
      >
        <Header
          className=""
        >
          <header
            className="c-header"
          >
            <div
              className="c-header__brand"
            >
              <Link
                className="ml-2 mr-2"
                onClick={[Function]}
                to="/"
              >
                <div
                  className="mock-link"
                />
              </Link>
              <div
                className="c-header__skillelinje"
              />
              <div
                className="c-header__tittel"
              >
                <span>
                  ui:app-name
                </span>
              </div>
            </div>
            <Undertittel>
              <n
                type="undertittel"
              >
                <h2
                  className="typo-undertittel"
                >
                  ui:app-title
                </h2>
              </n>
            </Undertittel>
            <div
              className="c-header__saksbehandler"
            >
              <Lenke
                className="c-header__highcontrast-link mr-3"
                href="#highContrast"
                onClick={[Function]}
              >
                <a
                  className="lenke c-header__highcontrast-link mr-3"
                  href="#highContrast"
                  onClick={[Function]}
                >
                  ui:label-highContrast
                </a>
              </Lenke>
            </div>
          </header>
        </Header>
        <Alert
          onClose={[Function]}
          type="server"
        />
        <main
          className="vedlegg _container p-0 container-fluid"
          id="main"
          role="main"
        >
          <Row
            className="m-0"
          >
            <div
              className="row m-0"
            >
              <div
                className="col-sm-2"
              />
              <div
                className="col-sm-8 m-4"
              >
                <Systemtittel
                  className="mt-4 mb-4"
                >
                  <n
                    className="mt-4 mb-4"
                    type="systemtittel"
                  >
                    <h2
                      className="typo-systemtittel mt-4 mb-4"
                    >
                      ui:title-vedlegg
                    </h2>
                  </n>
                </Systemtittel>
                <div
                  className="noSlideAnimate"
                >
                  <Input
                    bredde="fullbredde"
                    className="mb-4"
                    id="vedlegg-journalpostID-id"
                    label={
                      <div>
                        ui:label-journalpostID
                        <Hjelpetekst
                          id="journalPostID"
                        >
                          ui:help-journalpostID
                        </Hjelpetekst>
                      </div>
                    }
                    onChange={[Function]}
                  >
                    <div
                      className="skjemaelement mb-4"
                    >
                      <t
                        htmlFor="vedlegg-journalpostID-id"
                      >
                        <label
                          className="skjemaelement__label"
                          htmlFor="vedlegg-journalpostID-id"
                        >
                          <div>
                            ui:label-journalpostID
                            <Hjelpetekst
                              id="journalPostID"
                            >
                              <div
                                className="mock-hjelpetekst"
                              >
                                ui:help-journalpostID
                              </div>
                            </Hjelpetekst>
                          </div>
                        </label>
                      </t>
                      <input
                        aria-invalid={false}
                        className="skjemaelement__input input--fullbredde"
                        id="vedlegg-journalpostID-id"
                        onChange={[Function]}
                        type="text"
                      />
                    </div>
                  </Input>
                </div>
                <div
                  className="noSlideAnimate"
                  style={
                    Object {
                      "animationDelay": "0.15s",
                    }
                  }
                >
                  <Input
                    bredde="fullbredde"
                    className="mb-4"
                    id="vedlegg-dokumentID-id"
                    label={
                      <div>
                        ui:label-dokumentID
                        <Hjelpetekst
                          id="dokumentID"
                        >
                          ui:help-dokumentID
                        </Hjelpetekst>
                      </div>
                    }
                    onChange={[Function]}
                  >
                    <div
                      className="skjemaelement mb-4"
                    >
                      <t
                        htmlFor="vedlegg-dokumentID-id"
                      >
                        <label
                          className="skjemaelement__label"
                          htmlFor="vedlegg-dokumentID-id"
                        >
                          <div>
                            ui:label-dokumentID
                            <Hjelpetekst
                              id="dokumentID"
                            >
                              <div
                                className="mock-hjelpetekst"
                              >
                                ui:help-dokumentID
                              </div>
                            </Hjelpetekst>
                          </div>
                        </label>
                      </t>
                      <input
                        aria-invalid={false}
                        className="skjemaelement__input input--fullbredde"
                        id="vedlegg-dokumentID-id"
                        onChange={[Function]}
                        type="text"
                      />
                    </div>
                  </Input>
                </div>
                <div
                  className="noSlideAnimate"
                  style={
                    Object {
                      "animationDelay": "0.0s",
                    }
                  }
                >
                  <DocumentSearch
                    className="mb-4"
                    onDocumentFound={[Function]}
                    onRinasaksnummerChanged={[Function]}
                    resetValidation={[Function]}
                    validation={Object {}}
                  >
                    <div
                      className="mb-4 dokumentsok"
                    >
                      <div
                        className="dokumentsok__form"
                      >
                        <Input
                          bredde="fullbredde"
                          className="dokumentsok__form__input"
                          id="dokumentsok__form__input-id"
                          label="ui:label-rinasaksnummer"
                          onChange={[Function]}
                        >
                          <div
                            className="skjemaelement dokumentsok__form__input"
                          >
                            <t
                              htmlFor="dokumentsok__form__input-id"
                            >
                              <label
                                className="skjemaelement__label"
                                htmlFor="dokumentsok__form__input-id"
                              >
                                ui:label-rinasaksnummer
                              </label>
                            </t>
                            <input
                              aria-invalid={false}
                              className="skjemaelement__input input--fullbredde"
                              id="dokumentsok__form__input-id"
                              onChange={[Function]}
                              type="text"
                            />
                          </div>
                        </Input>
                        <Knapp
                          className="dokumentsok__knapp"
                          onClick={[Function]}
                        >
                          <KnappBase
                            autoDisableVedSpinner={false}
                            className="dokumentsok__knapp"
                            disabled={false}
                            htmlType="submit"
                            inaktivKlasseVedDisabled={false}
                            kompakt={false}
                            mini={false}
                            onClick={[Function]}
                            spinner={false}
                            type="standard"
                          >
                            <button
                              className="knapp dokumentsok__knapp knapp--standard"
                              disabled={false}
                              onClick={[Function]}
                              type="submit"
                            >
                              ui:form-search
                            </button>
                          </KnappBase>
                        </Knapp>
                      </div>
                      <div
                        className="dokumentsok__card mt-4 mb-4 slideAnimate"
                      >
                        <Select
                          bredde="fullbredde"
                          className="dokumentsok__card-select"
                          disabled={true}
                          id="dokumentsok__card-select-id"
                          label="ui:label-rinadokumentID"
                          onChange={[Function]}
                        >
                          <div
                            className="skjemaelement dokumentsok__card-select"
                          >
                            <t
                              htmlFor="dokumentsok__card-select-id"
                            >
                              <label
                                className="skjemaelement__label"
                                htmlFor="dokumentsok__card-select-id"
                              >
                                ui:label-rinadokumentID
                              </label>
                            </t>
                            <div
                              className="selectContainer input--fullbredde"
                            >
                              <select
                                aria-invalid={false}
                                className="skjemaelement__input"
                                disabled={true}
                                id="dokumentsok__card-select-id"
                                onChange={[Function]}
                              >
                                <option
                                  value=""
                                >
                                  ui:form-choose
                                </option>
                              </select>
                            </div>
                          </div>
                        </Select>
                      </div>
                    </div>
                  </DocumentSearch>
                </div>
                <div
                  className="vedlegg__submmit slideAnimate"
                  style={
                    Object {
                      "animationDelay": "0.45s",
                    }
                  }
                >
                  <Hovedknapp
                    disabled={false}
                    onClick={[Function]}
                    spinner={false}
                  >
                    <KnappBase
                      autoDisableVedSpinner={false}
                      className={null}
                      disabled={false}
                      htmlType="submit"
                      inaktivKlasseVedDisabled={false}
                      kompakt={false}
                      mini={false}
                      onClick={[Function]}
                      spinner={false}
                      type="hoved"
                    >
                      <button
                        className="knapp knapp--hoved"
                        disabled={false}
                        onClick={[Function]}
                        type="submit"
                      >
                        ui:label-sendVedlegg
                      </button>
                    </KnappBase>
                  </Hovedknapp>
                </div>
              </div>
              <div
                className="col-sm-2"
              />
            </div>
          </Row>
        </main>
        <SessionMonitor>
          <styled.div>
            <div
              className="sc-AxjAm kmXdIo"
            />
          </styled.div>
        </SessionMonitor>
        <Version />
      </ErrorBoundary>
    </UseErrorBoundaryWrapper>
  </TopContainer>
</Vedlegg>
`
